diff -urN zabbix-1.9.6.orig/src/zabbix_java/Makefile.in zabbix-1.9.6/src/zabbix_java/Makefile.in
--- zabbix-1.9.6.orig/src/zabbix_java/Makefile.in	2011-09-10 16:11:30.000000000 +0900
+++ zabbix-1.9.6/src/zabbix_java/Makefile.in	2011-09-28 17:07:55.889853286 +0900
@@ -420,14 +420,14 @@
 	mkdir -p class/tests
 
 install: all
-	mkdir -p $(sbindir)/zabbix_java/bin
-	mkdir -p $(sbindir)/zabbix_java/lib
-	cp startup.sh shutdown.sh $(sbindir)/zabbix_java
-	if test ! -f $(sbindir)/zabbix_java/settings.sh; then cp settings.sh $(sbindir)/zabbix_java; fi
-	cp $(ZJP) $(sbindir)/zabbix_java/bin
-	cp lib/*.jar $(sbindir)/zabbix_java/lib
-	if test ! -f $(sbindir)/zabbix_java/lib/logback.xml; then cp lib/logback.xml $(sbindir)/zabbix_java/lib; fi
-	if test ! -f $(sbindir)/zabbix_java/lib/logback-console.xml; then cp lib/logback-console.xml $(sbindir)/zabbix_java/lib; fi
+	mkdir -p $(DESTDIR)$(sbindir)/zabbix_java/bin
+	mkdir -p $(DESTDIR)$(sbindir)/zabbix_java/lib
+	cp startup.sh shutdown.sh $(DESTDIR)$(sbindir)/zabbix_java
+	if test ! -f $(DESTDIR)$(sbindir)/zabbix_java/settings.sh; then cp settings.sh $(DESTDIR)$(sbindir)/zabbix_java; fi
+	cp $(ZJP) $(DESTDIR)$(sbindir)/zabbix_java/bin
+	cp lib/*.jar $(DESTDIR)$(sbindir)/zabbix_java/lib
+	if test ! -f $(DESTDIR)$(sbindir)/zabbix_java/lib/logback.xml; then cp lib/logback.xml $(DESTDIR)$(sbindir)/zabbix_java/lib; fi
+	if test ! -f $(DESTDIR)$(sbindir)/zabbix_java/lib/logback-console.xml; then cp lib/logback-console.xml $(DESTDIR)$(sbindir)/zabbix_java/lib; fi
 
 clean:
 	rm -f $(ZJP)
