diff -Nru zabbix-1.8.orig/frontends/php/include/page_footer.php zabbix-1.8/frontends/php/include/page_footer.php
--- zabbix-1.8.orig/frontends/php/include/page_footer.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/include/page_footer.php	2009-12-14 21:52:54.000000000 +0900
@@ -78,9 +78,15 @@
 			$table->setCellSpacing(0);
 			$table->setCellPadding(1);
 			$table->addRow(array(
-				new CCol(new CLink(
+                                new CCol(array(
+                                        new CLink(
 					S_ZABBIX.SPACE.ZABBIX_VERSION.SPACE.S_COPYRIGHT_BY.SPACE.S_SIA_ZABBIX,
 					'http://www.zabbix.com', 'highlight', null, true),
+                                        new CSpan(SPACE.SPACE.'/'.SPACE.SPACE,'divider'),
+                                        new CLink(
+                                        'Powered by ZABBIX-JP',
+                                        'http://www.zabbix.jp','highlight', null, true) 
+                                        ),
 					'page_footer_l'),
 				new CCol(array(
 						new CSpan(SPACE.SPACE.'|'.SPACE.SPACE,'divider'),
diff -Nru zabbix-1.8.2.orig/frontends/php/include/page_header.php zabbix-1.8.2/frontends/php/include/page_header.php
--- zabbix-1.8.2.orig/frontends/php/include/page_header.php	2010-03-30 02:22:45.000000000 +0900
+++ zabbix-1.8.2/frontends/php/include/page_header.php	2010-05-06 11:32:27.000000000 +0900
@@ -184,10 +184,18 @@
 	if(!defined('ZBX_PAGE_NO_MENU')){
 COpt::compare_files_with_menu($ZBX_MENU);
 
+		if ($USER_DETAILS["lang"] == "ja_jp"){
+		$help = new CLink(S_HELP, 'http://www.zabbix.jp/documents', 'small_font', null, true);
+		$help->setTarget('_blank');
+		$support = new CLink(S_GET_SUPPORT, 'http://www.zabbix.jp/forum', 'small_font', null, true);
+		$support->setTarget('_blank');
+		}
+		else {
 		$help = new CLink(S_HELP, 'http://www.zabbix.com/documentation/', 'small_font', null, 'nosid');
 		$help->setTarget('_blank');
 		$support = new CLink(S_GET_SUPPORT, 'http://www.zabbix.com/support.php', 'small_font', null, 'nosid');
 		$support->setTarget('_blank');
+		}
 		$printview = new CLink(S_PRINT, $_SERVER['REQUEST_URI'].(empty($_GET)?'?':'&').'print=1', 'small_font');
 
 		$page_header_r_col = array($help,'|',$support,'|',$printview);
