diff -uNrp zabbix-1.8.2.orig/frontends/php/include/config.inc.php zabbix-1.8.2/frontends/php/include/config.inc.php
--- zabbix-1.8.2.orig/frontends/php/include/config.inc.php	2010-04-09 20:07:02.000000000 +0900
+++ zabbix-1.8.2/frontends/php/include/config.inc.php	2010-04-09 20:27:51.000000000 +0900
@@ -130,6 +130,8 @@
 		ob_start();
 		include $ZBX_CONFIGURATION_FILE;
 		ob_end_clean ();
+	}
+	if(file_exists($ZBX_CONFIGURATION_FILE) && isset($DB) && !isset($_COOKIE['ZBX_CONFIG']) && !isset($DENY_GUI)){
 		require_once('include/db.inc.php');
 
 		$error = '';
