diff -Nru zabbix-1.8.orig/frontends/php/include/locales/cn_zh.inc.php zabbix-1.8/frontends/php/include/locales/cn_zh.inc.php
--- zabbix-1.8.orig/frontends/php/include/locales/cn_zh.inc.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/include/locales/cn_zh.inc.php	2009-12-14 21:57:22.000000000 +0900
@@ -113,7 +113,6 @@
 	'S_LATEST_DATA_BIG'=>			'最新数据',
 	'S_ALL_SMALL'=>			'all',
 	'S_GRAPH'=>			'图形显示',
-	'S_COPYRIGHT_BY'=>			'Copyright 2001-2006 by',
 	'S_CONNECTED_AS'=>			'Connected as',
 	'S_SIA_ZABBIX'=>			'SIA Zabbix',
 	'S_ITEM_ADDED'=>			'Item added',
diff -Nru zabbix-1.8.orig/frontends/php/include/locales/de_de.inc.php zabbix-1.8/frontends/php/include/locales/de_de.inc.php
--- zabbix-1.8.orig/frontends/php/include/locales/de_de.inc.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/include/locales/de_de.inc.php	2009-12-14 21:57:29.000000000 +0900
@@ -392,7 +392,6 @@
 	'S_MINUS_ALL_MINUS'=>			'- alle -',
 	'S_MINUS_OTHER_MINUS'=>			'- andere -',
 	'S_GRAPH'=>			'Graph',
-	'S_COPYRIGHT_BY'=>			'Copyright 2001-2006 by',
 	'S_CONNECTED_AS'=>			'Angemeldet:',
 	'S_NOT_CONNECTED'=>			'Nicht angemeldet',
 	'S_SIA_ZABBIX'=>			'SIA Zabbix',
diff -Nru zabbix-1.8.orig/frontends/php/include/locales/fr_fr.inc.php zabbix-1.8/frontends/php/include/locales/fr_fr.inc.php
--- zabbix-1.8.orig/frontends/php/include/locales/fr_fr.inc.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/include/locales/fr_fr.inc.php	2009-12-14 21:57:39.000000000 +0900
@@ -464,7 +464,6 @@
 	'S_NOT_SELECTED_SMALL'=>			'non sélectionné',
 	'S_GRAPH'=>			'Graphique',
 	'S_ZABBIX'=>			'ZABBIX',
-	'S_COPYRIGHT_BY'=>			'Copyright 2001-2009 par',
 	'S_CONNECTED_AS'=>			'Connecté en tant que',
 	'S_NOT_CONNECTED'=>			'Non connecté',
 	'S_SIA_ZABBIX'=>			'SIA Zabbix',
diff -Nru zabbix-1.8.orig/frontends/php/include/locales/hu_hu.inc.php zabbix-1.8/frontends/php/include/locales/hu_hu.inc.php
--- zabbix-1.8.orig/frontends/php/include/locales/hu_hu.inc.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/include/locales/hu_hu.inc.php	2009-12-14 21:57:50.000000000 +0900
@@ -324,7 +324,6 @@
 	'S_MINUS_ALL_MINUS'=>			'- mind -',
 	'S_MINUS_OTHER_MINUS'=>			'- egyéb -',
 	'S_GRAPH'=>			'Grafikon',
-	'S_COPYRIGHT_BY'=>			'Copyright 2001-2007 by',
 	'S_CONNECTED_AS'=>			'Connected as',
 	'S_SIA_ZABBIX'=>			'SIA Zabbix',
 	'S_ITEM_ADDED'=>			'Item added',
diff -Nru zabbix-1.8.orig/frontends/php/include/locales/it_it.inc.php zabbix-1.8/frontends/php/include/locales/it_it.inc.php
--- zabbix-1.8.orig/frontends/php/include/locales/it_it.inc.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/include/locales/it_it.inc.php	2009-12-14 21:58:00.000000000 +0900
@@ -109,7 +109,6 @@
 	'S_NO_PERMISSIONS'=>			'Accesso negato!',
 	'S_ALL_SMALL'=>			'tutti',
 	'S_GRAPH'=>			'Graph',
-	'S_COPYRIGHT_BY'=>			'Copyright 2001-2006 by',
 	'S_CONNECTED_AS'=>			'Connesso come utente:',
 	'S_SIA_ZABBIX'=>			'SIA Zabbix',
 	'S_ITEM_ADDED'=>			'Elemento agguinto!',
diff -Nru zabbix-1.8.orig/frontends/php/include/locales/ja_jp.inc.php zabbix-1.8/frontends/php/include/locales/ja_jp.inc.php
--- zabbix-1.8.orig/frontends/php/include/locales/ja_jp.inc.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/include/locales/ja_jp.inc.php	2009-12-14 21:58:12.000000000 +0900
@@ -449,7 +449,6 @@
 	'S_NOT_SELECTED_SMALL'=>			'選択されていません',
 	'S_GRAPH'=>			'グラフ',
 	'S_ZABBIX'=>			'ZABBIX',
-	'S_COPYRIGHT_BY'=>			'Copyright 2001-2009 by',
 	'S_CONNECTED_AS'=>			'次のユーザでログイン中',
 	'S_NOT_CONNECTED'=>			'未ログイン',
 	'S_SIA_ZABBIX'=>			'SIA Zabbix',
diff -Nru zabbix-1.8.orig/frontends/php/include/locales/lv_lv.inc.php zabbix-1.8/frontends/php/include/locales/lv_lv.inc.php
--- zabbix-1.8.orig/frontends/php/include/locales/lv_lv.inc.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/include/locales/lv_lv.inc.php	2009-12-14 21:58:18.000000000 +0900
@@ -109,7 +109,6 @@
 	'S_NO_PERMISSIONS'=>			'No permissions !',
 	'S_ALL_SMALL'=>			'all',
 	'S_GRAPH'=>			'Graph',
-	'S_COPYRIGHT_BY'=>			'Copyright 2001-2006 by',
 	'S_CONNECTED_AS'=>			'Connected as',
 	'S_SIA_ZABBIX'=>			'SIA Zabbix',
 	'S_ITEM_ADDED'=>			'Item added',
diff -Nru zabbix-1.8.orig/frontends/php/include/locales/pl_pl.inc.php zabbix-1.8/frontends/php/include/locales/pl_pl.inc.php
--- zabbix-1.8.orig/frontends/php/include/locales/pl_pl.inc.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/include/locales/pl_pl.inc.php	2009-12-14 21:58:24.000000000 +0900
@@ -394,7 +394,6 @@
 	'S_MINUS_ALL_MINUS'=>				'- wszystkie -',
 	'S_MINUS_OTHER_MINUS'=>				'- inne -',
 	'S_GRAPH'=>					'Wykres',
-	'S_COPYRIGHT_BY'=>				'Prawa autorskie 2001-2009 dla',
 	'S_CONNECTED_AS'=>				'Połączony jako',
 	'S_NOT_CONNECTED'=>				'Nie połączony',
 	'S_SIA_ZABBIX'=>				'SIA Zabbix',
diff -Nru zabbix-1.8.orig/frontends/php/include/locales/sp_sp.inc.php zabbix-1.8/frontends/php/include/locales/sp_sp.inc.php
--- zabbix-1.8.orig/frontends/php/include/locales/sp_sp.inc.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/include/locales/sp_sp.inc.php	2009-12-14 21:58:30.000000000 +0900
@@ -324,7 +324,6 @@
 	'S_MINUS_ALL_MINUS'=>			'- todo -',
 	'S_MINUS_OTHER_MINUS'=>			'- otros -',
 	'S_GRAPH'=>			'Gráficos',
-	'S_COPYRIGHT_BY'=>			'Copyright 2001-2007 por',
 	'S_CONNECTED_AS'=>			'Conectado como',
 	'S_SIA_ZABBIX'=>			'SIA Zabbix',
 	'S_ITEM_ADDED'=>			'Monitor añadido',
