diff -uNr zabbix-1.6.4.orig/frontends/php/include/classes/graph.inc.php zabbix-1.6.4/frontends/php/include/classes/graph.inc.php
--- zabbix-1.6.4.orig/frontends/php/include/classes/graph.inc.php	2009-04-07 20:41:04.000000000 +0900
+++ zabbix-1.6.4/frontends/php/include/classes/graph.inc.php	2009-04-08 13:27:06.000000000 +0900
@@ -250,6 +250,8 @@
 	}
 
 	function drawHeader(){
+		global $USER_DETAILS;
+
 		if(!isset($this->header)){
 			$str=$this->items[0]['host'].':'.$this->items[0]['description'];
 		}
@@ -266,7 +268,19 @@
 			$fontnum = 4;
 		}
 		$x=$this->fullSizeX/2-imagefontwidth($fontnum)*strlen($str)/2;
-		imagestring($this->im, $fontnum,$x,1, $str , $this->GetColor('Dark Red No Alpha'));
+		//imagestring($this->im, $fontnum,$x,1, $str , $this->GetColor('Dark Red No Alpha'));
+		if ($USER_DETAILS["lang"] == "ja_jp"){
+			ImageTTFText($this->im, $fontnum+6, 0, $x+10, 13, $this->GetColor("Dark Red No Alpha"), "/usr/share/fonts/japanese/TrueType/sazanami-gothic.ttf", $str);
+		}
+		elseif ($USER_DETAILS["lang"] == "cn_zh"){
+			ImageTTFText($this->im, $fontnum+6, 0, $x+10, 13, $this->GetColor("Dark Red No Alpha"), "/usr/share/fonts/chinese/TrueType/fireflysung.ttf", $str);
+		}
+		elseif ($USER_DETAILS["lang"] == "ko_kr"){
+			ImageTTFText($this->im, $fontnum+6, 0, $x+10, 13, $this->GetColor("Dark Red No Alpha"), "/usr/share/fonts/korean/gulim.ttf", $str);
+		}
+		else {
+			imagestring($this->im, $fontnum, $x, 1, $str, $this->GetColor("Dark Red No Alpha"));
+		}
 	}
 
 	function setHeader($header){
diff -uNr zabbix-1.6.4.orig/frontends/php/map.php zabbix-1.6.4/frontends/php/map.php
--- zabbix-1.6.4.orig/frontends/php/map.php	2009-04-07 20:41:04.000000000 +0900
+++ zabbix-1.6.4/frontends/php/map.php	2009-04-08 11:58:42.000000000 +0900
@@ -97,7 +97,19 @@
 	}
 	else{
 		$x=imagesx($im)/2-ImageFontWidth(4)*strlen($name)/2;
-		imagestring($im, 4,$x,1, $name , $darkred);
+		//imagestring($im, 4,$x,1, $name , $darkred);
+		if ($USER_DETAILS["lang"] == "ja_jp"){
+			ImageTTFText($im, 10, 0, $x+10, 14, $darkred, "/usr/share/fonts/japanese/TrueType/sazanami-gothic.ttf", $name);
+		}
+		elseif ($USER_DETAILS["lang"] == "cn_zh") {
+			ImageTTFText($im, 10, 0, $x+10, 14, $darkred, "/usr/share/fonts/chinese/TrueType/fireflysung.ttf", $name); 
+		}
+		elseif ($USER_DETAILS["lang"] == "ko_kr"){
+			ImageTTFText($im, 10, 0, $x+10, 14, $darkred, "/usr/share/fonts/korean/TrueType/gulim.ttf", $name); 
+		}
+		else {
+			imagestring($im, 4,$x,1, $name , $darkred);
+		}
 	}
 	unset($db_image);
 
@@ -241,7 +253,19 @@
 				$x_label-2, $y_label,
 				$x_label+ImageFontWidth(2)*strlen($label_line), $y_label+ImageFontHeight(2),
 				$white);
-			imagestring($im, 2, $x_label, $y_label, $label_line,$label_color);
+			//imagestring($im, 2, $x_label, $y_label, $label_line,$label_color);
+			if ($USER_DETAILS["lang"] == "ja_jp"){
+				ImageTTFText($im, 9, 0, $x_label+2, $y_label+10, $label_color, "/usr/share/fonts/japanese/TrueType/sazanami-gothic.ttf", $label_line);
+				}
+			elseif ($USER_DETAILS["lang"] == "cn_zh"){
+				ImageTTFText($im, 9, 0, $x_label+2, $y_label+10, $label_color, "/usr/share/fonts/chinese/TrueType/fireflysung.ttf", $label_line); 
+				}
+			elseif ($USER_DETAILS["lang"] == "ko_kr"){
+                                ImageTTFText($im, 9, 0, $x_label+2, $y_label+10, $label_color, "/usr/share/fonts/korean/TrueType/gulim.ttf", $label_line);
+                                }
+			else {
+				imagestring($im, 2, $x_label, $y_label, $label_line, $label_color);
+			}
 		}
 
 		if($info_line!=''){
@@ -249,7 +273,19 @@
 				$x_info-2, $y_info,
 				$x_info+ImageFontWidth(2)*strlen($info_line), $y_info+ImageFontHeight(2),
 				$white);
-			imagestring($im, 2, $x_info, $y_info, $info_line,$color);
+			//imagestring($im, 2, $x_info, $y_info, $info_line,$color);
+			if ($USER_DETAILS["lang"] == "ja_jp"){
+				ImageTTFText($im, 8, 0, $x_info+9, $y_info+10, $color, "/usr/share/fonts/japanese/TrueType/sazanami-gothic.ttf", $info_line);
+			}
+			elseif ($USER_DETAILS["lang"] == "cn_zh"){
+				ImageTTFText($im, 8, 0, $x_info+9, $y_info+10, $color, "/usr/share/fonts/chinese/TrueType/fireflysung.ttf", $info_line);
+			}
+			elseif ($USER_DETAILS["lang"] == "ko_kr"){
+				ImageTTFText($im, 8, 0, $x_info+9, $y_info+10, $color, "/usr/share/fonts/korean/TrueType/gulim.ttf", $info_line);
+			}
+			else {
+				imagestring($im, 2, $x_info, $y_info, $info_line,$color);
+			}
 		}
 	}
 
