--- zabbix-1.4.5.orig/src/libs/zbxcommon/comms.c	2008-03-26 05:14:02.000000000 +0900
+++ zabbix-1.4.5/src/libs/zbxcommon/comms.c	2009-05-18 10:37:42.000000000 +0900
@@ -51,7 +51,7 @@
 {
 #define ADD_XML_DATA(tag_name, var) \
 	data_b64[0] = '\0'; \
-	str_base64_encode(var, data_b64, (int)strlen(var)); \
+	str_base64_encode(var, data_b64, (int)strlen(var) >= ZBX_MAX_B64_LEN/4*3 ? ZBX_MAX_B64_LEN/4*3-3 : (int)strlen(var)); \
 	request = zbx_strdcatf(request, "<" tag_name ">%s</" tag_name ">",	data_b64)
 
 	char data_b64[ZBX_MAX_B64_LEN];

