diff -urN zabbix-2.0.0rc3.orig/frontends/php/include/page_footer.php zabbix-2.0.0rc3/frontends/php/include/page_footer.php
--- zabbix-2.0.0rc3.orig/frontends/php/include/page_footer.php	2012-04-20 18:12:54.000000000 +0900
+++ zabbix-2.0.0rc3/frontends/php/include/page_footer.php	2012-04-21 03:03:07.405940136 +0900
@@ -99,9 +99,15 @@
 		}
 
 		$table->addRow(array(
-			new CCol(new CLink(
+			new CCol(array(
+				new CLink(
 				_s('Zabbix %s Copyright 2001-2012 by Zabbix SIA', ZABBIX_VERSION),
-				'http://www.zabbix.com', 'highlight', null, true), 'center'),
+				'http://www.zabbix.com', 'highlight', null, true),
+				new CSpan(SPACE.SPACE.'|'.SPACE.SPACE, 'divider'),
+				new CLink(
+				'Powered by ZABBIX-JP',
+				'http://www.zabbix.jp', 'highlight', null, true),
+				), 'center'),
 			new CCol(array(
 				new CSpan(SPACE.SPACE.'|'.SPACE.SPACE, 'divider'),
 				new CSpan($conString, 'footer_sign')
diff -urN zabbix-2.0.0rc3.orig/frontends/php/include/page_header.php zabbix-2.0.0rc3/frontends/php/include/page_header.php
--- zabbix-2.0.0rc3.orig/frontends/php/include/page_header.php	2012-04-20 18:12:54.000000000 +0900
+++ zabbix-2.0.0rc3/frontends/php/include/page_header.php	2012-04-21 02:58:03.888035194 +0900
@@ -228,10 +228,18 @@
 }
 
 if (!defined('ZBX_PAGE_NO_MENU')) {
+	if (CWebUser::$data["lang"] == "ja_JP"){
+	$help = new CLink(_('Help'), 'http://www.zabbix.jp/documents', 'small_font', null, 'nosid');
+	$help->setTarget('_blank');
+	$support = new CLink(_('Get support'), 'http://www.zabbix.jp/forum', 'small_font', null, 'nosid');
+	$support->setTarget('_blank');
+	}
+	else {
 	$help = new CLink(_('Help'), 'http://www.zabbix.com/documentation/', 'small_font', null, 'nosid');
 	$help->setTarget('_blank');
 	$support = new CLink(_('Get support'), 'http://www.zabbix.com/support.php', 'small_font', null, 'nosid');
 	$support->setTarget('_blank');
+	}
 
 	$req = new CUrl($_SERVER['REQUEST_URI']);
 	$req->setArgument('print', 1);
