diff -urN zabbix-2.0.0rc3.orig/frontends/php/js/functions.js zabbix-2.0.0rc3/frontends/php/js/functions.js
--- zabbix-2.0.0rc3.orig/frontends/php/js/functions.js	2012-04-20 18:12:53.000000000 +0900
+++ zabbix-2.0.0rc3/frontends/php/js/functions.js	2012-04-21 02:47:18.676965682 +0900
@@ -537,16 +537,16 @@
 	if (id != 0) {
 		show_popup_menu(evnt, [
 				[name, null, null, {'outer' : ['pum_oheader'], 'inner' : ['pum_iheader']}],
-				['Add Service', "javascript: window.location.href = 'services.php?form=1&parentid=" + id + "&parentname=" + name + "';", null, {'outer' : ['pum_o_item'], 'inner' : ['pum_i_item']}],
-				['Edit Service', "javascript: window.location.href = 'services.php?form=1&serviceid=" + id + "';", null, {'outer' : ['pum_o_item'], 'inner' : ['pum_i_item']}],
-				['Delete Service', "javascript: if (Confirm('Delete selected services?')) { window.location.href = 'services.php?delete=1&serviceid=" + id + "'; }", null, {'outer' : ['pum_o_item'], 'inner' : ['pum_i_item']}]
+				['サービスの追加', "javascript: window.location.href = 'services.php?form=1&parentid=" + id + "&parentname=" + name + "';", null, {'outer' : ['pum_o_item'], 'inner' : ['pum_i_item']}],
+				['サービスの変更', "javascript: window.location.href = 'services.php?form=1&serviceid=" + id + "';", null, {'outer' : ['pum_o_item'], 'inner' : ['pum_i_item']}],
+				['サービスの削除', "javascript: if (Confirm('Delete selected services?')) { window.location.href = 'services.php?delete=1&serviceid=" + id + "'; }", null, {'outer' : ['pum_o_item'], 'inner' : ['pum_i_item']}]
 			],
 			120
 		);
 	} else {
 		show_popup_menu(evnt, [
 				[name, null, null, {'outer' : ['pum_oheader'], 'inner' : ['pum_iheader']}],
-				['Add Service',"javascript: window.location.href = 'services.php?form=1&parentid=" + id + "&parentname=" + name + "';", null, {'outer' : ['pum_o_item'], 'inner' : ['pum_i_item']}]
+				['サービスの追加',"javascript: window.location.href = 'services.php?form=1&parentid=" + id + "&parentname=" + name + "';", null, {'outer' : ['pum_o_item'], 'inner' : ['pum_i_item']}]
 			],
 			120
 		);
