diff -Nru zabbix-1.6.6.orig/frontends/php/include/locales/cn_zh.inc.php zabbix-1.6.6/frontends/php/include/locales/cn_zh.inc.php
--- zabbix-1.6.6.orig/frontends/php/include/locales/cn_zh.inc.php	2009-08-26 22:16:26.000000000 +0900
+++ zabbix-1.6.6/frontends/php/include/locales/cn_zh.inc.php	2009-10-28 00:16:39.000000000 +0900
@@ -453,7 +453,7 @@
 	'S_LATVIAN_LV'=>			'Latvian (LV)',
 	'S_RUSSIAN_RU'=>			'Russian (RU)',
 	'S_SPANISH_SP'=>			'Spanish (SP)',
-	'S_JAPANESE_JP'=>			'Chinese (CN)',
+	'S_JAPANESE_JP'=>			'Japanese (JP)',
 	'S_ZABBIX_BIG'=>			'ZABBIX',
 	'S_HOST_PROFILES'=>			'Host profiles',
 	'S_HOST_PROFILES_BIG'=>			'请选择主机所在位置=>',
