diff -Nur zabbix-1.4.7.org/frontends/php/tr_events.php zabbix-1.4.7/frontends/php/tr_events.php
--- zabbix-1.4.7.org/frontends/php/tr_events.php	2011-01-03 08:11:03.000000000 -0800
+++ zabbix-1.4.7/frontends/php/tr_events.php	2011-01-21 00:33:48.000000000 -0800
@@ -96,22 +96,19 @@
 	$table->ShowStart();
 
 	$rows = array();
-	$count = 0;
-	while($row=DBfetch($result)){
-		if(!empty($rows) && $rows[$count]['value'] != $row['value']){
-			$count++;
-		}
-		$rows[$count] = $row;
-	}
 
 	$truesum=0;
 	$falsesum=0;
 	$dissum=0;
 	$clock=time();
-	
-//	while($row=DBfetch($result))
-	foreach($rows as $id => $row)
-	{
+	$row_pre=null;
+	while($row=DBfetch($result)){
+		if(!empty($row_pre) && $row_pre['value'] == $row['value']){
+			$row_pre = $row;
+			continue;
+		}
+		$row_pre = $row;
+
 		$lclock=$clock;
 		$clock=$row["clock"];
 		$leng=$lclock-$row["clock"];
