diff -Nur zabbix-1.4.7.org/frontends/php/include/classes/graph.inc.php zabbix-1.4.7/frontends/php/include/classes/graph.inc.php
--- zabbix-1.4.7.org/frontends/php/include/classes/graph.inc.php	2011-01-03 08:11:03.000000000 -0800
+++ zabbix-1.4.7/frontends/php/include/classes/graph.inc.php	2011-01-20 22:46:40.000000000 -0800
@@ -595,6 +595,8 @@
 		
 		function drawLegend()
 		{
+			global $USER_DETAILS;
+
 			$max_host_len=0;
 			$max_desc_len=0;
 			for($i=0;$i<$this->num;$i++)
@@ -669,13 +671,43 @@
 					6,
 					$this->GetColor("Black No Alpha"));
 
-				ImageString(
-					$this->im, 
-					2,
-					$this->shiftXleft+9,
-					$this->sizeY+$this->shiftY+(62-5)+12*$i,
-					$trigger['description'],
-					$this->GetColor("Black No Alpha"));
+				if ($USER_DETAILS["lang"] == "ja_jp")
+				{
+					ImageTTFText($this->im, 8, 0,
+						$this->shiftXleft+9,
+						$this->sizeY+$this->shiftY+(62-5)+12*$i+11,
+						$this->GetColor("Black No Alpha"),
+						"/usr/share/fonts/japanese/TrueType/sazanami-gothic.ttf",
+						$trigger['description']);
+				}
+				elseif ($USER_DETAILS["lang"] == "cn_zh")
+				{
+					ImageTTFText($this->im, 8, 0,
+						$this->shiftXleft+9,
+						$this->sizeY+$this->shiftY+(62-5)+12*$i+11,
+						$this->GetColor("Black No Alpha"),
+						"/usr/share/fonts/chinese/TrueType/zysong.ttf",
+						$trigger['description']);
+				}
+				elseif ($USER_DETAILS["lang"] == "ko_kr")
+				{
+					ImageTTFText($this->im, 8, 0,
+						$this->shiftXleft+9,
+						$this->sizeY+$this->shiftY+(62-5)+12*$i+11,
+						$this->GetColor("Black No Alpha"),
+						"/usr/share/fonts/korean/TrueType/gulim.ttf",
+						$trigger['description']);
+				}
+				else
+				{
+					ImageString(
+						$this->im, 
+						2,
+						$this->shiftXleft+9,
+						$this->sizeY+$this->shiftY+(62-5)+12*$i,
+						$trigger['description'],
+						$this->GetColor("Black No Alpha"));
+				}
 				++$i;
 			}
 		}
