diff -Nru zabbix-1.8.orig/frontends/php/include/locales/en_gb.inc.php zabbix-1.8/frontends/php/include/locales/en_gb.inc.php
--- zabbix-1.8.orig/frontends/php/include/locales/en_gb.inc.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/include/locales/en_gb.inc.php	2009-12-14 21:46:23.000000000 +0900
@@ -1348,6 +1348,7 @@
 	'S_CANNOT_ADD_SERVICE'=>		'Cannot add service',
 	'S_SERVICE_DELETED'=>			'Service deleted',
 	'S_CANNOT_DELETE_SERVICE'=>		'Cannot delete service',
+	'S_PARENT_SERVICE'=>			'Parent Service',
 	'S_STATUS_CALCULATION'=>		'Status calculation',
 	'S_STATUS_CALCULATION_ALGORITHM'=>	'Status calculation algorithm',
 	'S_NONE'=>				'None',
diff -Nru zabbix-1.8.orig/frontends/php/services_form.php zabbix-1.8/frontends/php/services_form.php
--- zabbix-1.8.orig/frontends/php/services_form.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/services_form.php	2009-12-14 21:45:20.000000000 +0900
@@ -475,7 +475,7 @@
 	$cb->setType('button');
 	$cb->setAction("javascript: openWinCentered('services_form.php?pservices=1".url_param('serviceid')."','ZBX_Services_List',740,420,'scrollbars=1, toolbar=0, menubar=0, resizable=1, dialog=0');");
 
-	$frmService->addRow('Parent Service',array($ctb,$cb));
+	$frmService->addRow(S_PARENT_SERVICE,array($ctb,$cb));
 //----------
 
 //child links
