diff -Nru zabbix-1.8.orig/frontends/php/popup_media.php zabbix-1.8/frontends/php/popup_media.php
--- zabbix-1.8.orig/frontends/php/popup_media.php	2009-12-08 04:43:57.000000000 +0900
+++ zabbix-1.8/frontends/php/popup_media.php	2010-01-17 21:48:10.000000000 +0900
@@ -141,7 +141,7 @@
 	$cmbStat = new CComboBox('active',$active);
 	$cmbStat->addItem(0,S_ENABLED);
 	$cmbStat->addItem(1,S_DISABLED);
-	$frmMedia->addRow('Status',$cmbStat);
+	$frmMedia->addRow(S_STATUS,$cmbStat);
 
 	$frmMedia->addItemToBottomRow(new CButton('add', ($media > -1)?S_SAVE:S_ADD));
 	$frmMedia->addItemToBottomRow(SPACE);
