--- zabbix-1.4.6.orig/frontends/php/map.php	2009-03-07 13:47:01.000000000 +0900
+++ zabbix-1.4.6/frontends/php/map.php	2009-03-09 20:51:36.000000000 +0900
@@ -77,18 +77,18 @@ include_once "include/page_header.php";
 	$black		= ImageColorAllocate($im,0,0,0); 
 	$gray		= ImageColorAllocate($im,150,150,150);
 
-	$colors["Red"]		= ImageColorAllocate($im,255,0,0); 
-	$colors["Dark Red"]	= ImageColorAllocate($im,150,0,0); 
-	$colors["Green"]	= ImageColorAllocate($im,0,255,0); 
-	$colors["Dark Green"]	= ImageColorAllocate($im,0,150,0); 
-	$colors["Blue"]		= ImageColorAllocate($im,0,0,255); 
-	$colors["Dark Blue"]	= ImageColorAllocate($im,0,0,150); 
-	$colors["Yellow"]	= ImageColorAllocate($im,255,255,0); 
-	$colors["Dark Yellow"]	= ImageColorAllocate($im,150,150,0); 
-	$colors["Cyan"]		= ImageColorAllocate($im,0,255,255); 
-	$colors["Black"]	= ImageColorAllocate($im,0,0,0); 
-	$colors["Gray"]		= ImageColorAllocate($im,150,150,150); 
-	$colors["White"]	= ImageColorAllocate($im,255,255,255);
+	$colors[S_RED]          = ImageColorAllocate($im,255,0,0);
+	$colors[S_DARK_RED]     = ImageColorAllocate($im,150,0,0);
+	$colors[S_GREEN]        = ImageColorAllocate($im,0,255,0);
+	$colors[S_DARK_GREEN]   = ImageColorAllocate($im,0,150,0);
+	$colors[S_BLUE]         = ImageColorAllocate($im,0,0,255);
+	$colors[S_DARK_BLUE]    = ImageColorAllocate($im,0,0,150);
+	$colors[S_YELLOW]       = ImageColorAllocate($im,255,255,0);
+	$colors[S_DARK_YELLOW]  = ImageColorAllocate($im,150,150,0);
+	$colors[S_CYAN]         = ImageColorAllocate($im,0,255,255);
+	$colors[S_BLACK]        = ImageColorAllocate($im,0,0,0);
+	$colors[S_GRAY]         = ImageColorAllocate($im,150,150,150);
+	$colors[S_WHITE]        = ImageColorAllocate($im,255,255,255);
 
 	$x=imagesx($im); 
 	$y=imagesy($im);
@@ -301,7 +301,7 @@ include_once "include/page_header.php";
 
 	if(!isset($_REQUEST["border"]))
 	{
-		ImageRectangle($im,0,0,$width-1,$height-1,$colors["Black"]);
+		ImageRectangle($im,0,0,$width-1,$height-1,$colors[S_BLACK]);
 	}
 
 	ImageOut($im);
--- zabbix-1.4.6.orig/frontends/php/include/maps.inc.php	2008-07-23 20:47:31.000000000 +0900
+++ zabbix-1.4.6/frontends/php/include/maps.inc.php	2009-03-09 20:55:14.000000000 +0900
@@ -513,9 +513,9 @@
 				$out['info'] = $inf['info'];
 
 			if($inf['priority'] > 3)
-				$out['color'] = $colors['Red'];
+				$out['color'] = $colors[S_RED];
 			else
-				$out['color'] = $colors['Dark Red'];
+				$out['color'] = $colors[S_DARK_RED];
 
 			$out['iconid'] = $db_element['iconid_on'];
 		}
@@ -531,7 +531,7 @@
 			else */ if(isset($inf['info']))
 				$out['info'] = $inf['info'];
 
-			$out['color'] = $colors['Gray'];
+			$out['color'] = $colors[S_GRAY];
 			$out['iconid'] = $db_element['iconid_unknown'];
 		}
 		else
@@ -544,7 +544,7 @@
 			if(isset($inf['info']))
 				$out['info'] = S_OK_BIG;
 
-			$out['color'] = $colors['Dark Green'];
+			$out['color'] = $colors[S_DARK_GREEN];
 			$out['iconid'] = $db_element['iconid_off'];
 		}
 
--- zabbix-1.4.6.orig/frontends/php/include/forms.inc.php	2009-03-07 13:47:01.000000000 +0900
+++ zabbix-1.4.6/frontends/php/include/forms.inc.php	2009-03-09 21:45:25.000000000 +0900
@@ -4770,7 +4770,7 @@ include_once 'include/discovery.inc.php'
 		/*foreach(array('Black','Blue','Cyan','Dark Blue','Dark Green',
 			'Dark Red','Dark Yellow','Green','Red','White','Yellow') as $value) */
 		foreach(array(S_BLACK,S_BLUE,S_CYAN,S_DARK_BLUE,S_DARK_GREEN,
-			S_DARK_RED,S_DARK_YELLOW,S_GREEN,S_RED,S_WHITE,S_YELLOW) as $value)
+			S_DARK_RED,S_DARK_YELLOW,S_GRAY,S_GREEN,S_RED,S_WHITE,S_YELLOW) as $value)
 		{
 			$cmbColor_off->AddItem($value, $value);
 			$cmbColor_on->AddItem($value, $value);
--- zabbix-1.4.6.orig/frontends/php/include/locales/en_gb.inc.php	2009-03-07 13:47:01.000000000 +0900
+++ zabbix-1.4.6/frontends/php/include/locales/en_gb.inc.php	2009-03-09 20:42:54.000000000 +0900
@@ -362,6 +362,7 @@
 	'S_DARK_GREEN'=>			'Dark green',
 	'S_DARK_RED'=>				'Dark red',
 	'S_DARK_YELLOW'=>			'Dark yellow',
+	'S_GRAY'=>				'Gray',
 	'S_GREEN'=>				'Green',
 	'S_RED'=>				'Red',
 	'S_WHITE'=>				'White',
--- zabbix-1.4.6.orig/frontends/php/include/locales/ja_jp.inc.php	2009-03-07 13:47:02.000000000 +0900
+++ zabbix-1.4.6/frontends/php/include/locales/ja_jp.inc.php	2009-03-09 20:50:23.000000000 +0900
@@ -362,6 +362,7 @@
 	'S_DARK_GREEN'=>			'濃い緑',
 	'S_DARK_RED'=>				'濃い赤',
 	'S_DARK_YELLOW'=>			'濃い黄',
+	'S_GRAY'=>				'グレイ',
 	'S_GREEN'=>				'緑',
 	'S_RED'=>				'赤',
 	'S_WHITE'=>				'白',
--- zabbix-1.4.6.orig/frontends/php/include/locales/cn_zh.inc.php	2009-03-07 13:47:02.000000000 +0900
+++ zabbix-1.4.6/frontends/php/include/locales/cn_zh.inc.php	2009-03-09 20:47:25.000000000 +0900
@@ -126,5 +126,6 @@
 	"S_DARK_GREEN"=>			"Dark green",
 	"S_DARK_RED"=>				"Dark red",
 	"S_DARK_YELLOW"=>			"Dark yellow",
+	'S_GRAY'=>				"Gray",
 	"S_GREEN"=>				"Green",
 	"S_RED"=>				"Red",
 	"S_WHITE"=>				"White",