diff -Nur zabbix-1.4.7.org/create/data/data.sql zabbix-1.4.7/create/data/data.sql
--- zabbix-1.4.7.org/create/data/data.sql	2011-01-03 08:11:03.000000000 -0800
+++ zabbix-1.4.7/create/data/data.sql	2011-01-20 20:50:00.000000000 -0800
@@ -35,8 +35,8 @@
 -- Dumping data for table `users`
 --
 
-INSERT INTO users VALUES (1,'Admin','Zabbix','Administrator','d41d8cd98f00b204e9800998ecf8427e','',900,'en_gb',30,3);
-INSERT INTO users VALUES (2,'guest','Default','User','d41d8cd98f00b204e9800998ecf8427e','',900,'en_gb',30,1);
+INSERT INTO users VALUES (1,'Admin','Zabbix','Administrator','d41d8cd98f00b204e9800998ecf8427e','',900,'ja_jp',30,3);
+INSERT INTO users VALUES (2,'guest','Default','User','d41d8cd98f00b204e9800998ecf8427e','',900,'ja_jp',30,1);
 
 --
 -- Dumping data for table `usrgrp`
