diff -Nur zabbix-1.4.7.org/src/libs/zbxcommon/comms.c zabbix-1.4.7/src/libs/zbxcommon/comms.c
--- zabbix-1.4.7.org/src/libs/zbxcommon/comms.c	2011-01-03 08:11:02.000000000 -0800
+++ zabbix-1.4.7/src/libs/zbxcommon/comms.c	2011-01-20 21:33:16.000000000 -0800
@@ -51,7 +51,7 @@
 {
 #define ADD_XML_DATA(tag_name, var) \
 	data_b64[0] = '\0'; \
-	str_base64_encode(var, data_b64, (int)strlen(var)); \
+	str_base64_encode(var, data_b64, (int)strlen(var) >= ZBX_MAX_B64_LEN/4*3 ? ZBX_MAX_B64_LEN/4*3-3 : (int)strlen(var)); \
 	request = zbx_strdcatf(request, "<" tag_name ">%s</" tag_name ">",	data_b64)
 
 	char data_b64[ZBX_MAX_B64_LEN];
