diff -Nur zabbix-1.4.7.org/frontends/php/include/events.inc.php zabbix-1.4.7/frontends/php/include/events.inc.php
--- zabbix-1.4.7.org/frontends/php/include/events.inc.php	2011-01-03 08:11:03.000000000 -0800
+++ zabbix-1.4.7/frontends/php/include/events.inc.php	2011-01-20 22:04:40.000000000 -0800
@@ -51,7 +51,6 @@
 		}
 		else
 		{
-			$sql_from = ", hosts_groups hg ";
 			$sql_cond = " and h.hostid in (".$availiable_hosts.") ";
 		}
 	
