diff -Nur zabbix-1.4.7.org/frontends/php/include/page_footer.php zabbix-1.4.7/frontends/php/include/page_footer.php
--- zabbix-1.4.7.org/frontends/php/include/page_footer.php	2011-01-03 08:11:03.000000000 -0800
+++ zabbix-1.4.7/frontends/php/include/page_footer.php	2011-01-21 00:19:15.000000000 -0800
@@ -59,11 +59,17 @@
 			$table->SetCellSpacing(0);
 			$table->SetCellPadding(1);
 			$table->AddRow(array(
-				new CCol(new CLink(
+				new CCol(array(
+					new CLink(
 					S_ZABBIX_VER.SPACE.S_COPYRIGHT_BY.SPACE.S_SIA_ZABBIX,
 					"http://www.zabbix.com", "highlight"),
+					new CSpan(SPACE.SPACE.'/'.SPACE.SPACE,'divider'),
+					new CLink(
+						'Powered by ZABBIX-JP',
+						'http://www.zabbix.jp','highlight')
+					),
 					"page_footer_l"),
-				new CCol(array(
+					new CCol(array(
 						new CSpan(SPACE.SPACE."|".SPACE.SPACE,"divider"),
 						S_CONNECTED_AS.SPACE."'".$USER_DETAILS["alias"]."'".
 						(ZBX_DISTRIBUTED ? SPACE.S_FROM_SMALL.SPACE."'".$USER_DETAILS["node"]['name']."'" : '')
