diff -Nur zabbix-1.4.7.org/frontends/php/httpconf.php zabbix-1.4.7/frontends/php/httpconf.php
--- zabbix-1.4.7.org/frontends/php/httpconf.php	2011-01-03 08:11:03.000000000 -0800
+++ zabbix-1.4.7/frontends/php/httpconf.php	2011-01-20 23:06:19.000000000 -0800
@@ -476,7 +476,7 @@
 			$status=new CCol(new CLink(httptest_status2str($httptest_data["status"]),
 					"?group_httptestid%5B%5D=".$httptest_data["httptestid"].
 					"&hostid=".$_REQUEST["hostid"].
-					"&group_task=".($httptest_data["status"] ? "Activate+selected" : "Disable+selected"),
+					"&group_task=".($httptest_data["status"] ? S_ACTIVATE_SELECTED : S_DISABLE_SELECTED),
 					httptest_status2style($httptest_data["status"])));
 	
 
diff -Nur zabbix-1.4.7.org/frontends/php/items.php zabbix-1.4.7/frontends/php/items.php
--- zabbix-1.4.7.org/frontends/php/items.php	2011-01-03 08:11:03.000000000 -0800
+++ zabbix-1.4.7/frontends/php/items.php	2011-01-20 23:03:47.000000000 -0800
@@ -778,7 +778,7 @@
 
 			$status=new CCol(new CLink(item_status2str($db_item["status"]),
 					"?group_itemid%5B%5D=".$db_item["itemid"].
-					"&group_task=".($db_item["status"] ? "Activate+selected" : "Disable+selected"),
+					"&group_task=".($db_item["status"] ? S_ACTIVATE_SELECTED : S_DISABLE_SELECTED),
 					item_status2style($db_item["status"])));
 	
 			if($db_item["error"] == '')
