diff -Nur zabbix-1.4.7.org/frontends/php/include/defines.inc.php zabbix-1.4.7/frontends/php/include/defines.inc.php
--- zabbix-1.4.7.org/frontends/php/include/defines.inc.php	2011-01-03 08:11:03.000000000 -0800
+++ zabbix-1.4.7/frontends/php/include/defines.inc.php	2011-01-21 00:49:10.000000000 -0800
@@ -418,7 +418,7 @@
 	define('ZBX_SIMPLE_EXPRESSION_FUNCTION_NAME_ID', 8);
 	define('ZBX_SIMPLE_EXPRESSION_FUNCTION_PARAM_ID', 10);
 
-	define('ZBX_EREG_EXPRESSION_TOKEN_FORMAT', '^([[:print:]]*)('.ZBX_EREG_SIMPLE_EXPRESSION_FORMAT.'|'.ZBX_EREG_MACRO_NAME_FORMAT.')([[:print:]]*)$');
+	define('ZBX_EREG_EXPRESSION_TOKEN_FORMAT', '^([^[:cntrl:]]*)('.ZBX_EREG_SIMPLE_EXPRESSION_FORMAT.'|'.ZBX_EREG_MACRO_NAME_FORMAT.')([^[:cntrl:]]*)$');
 
 	define('ZBX_EXPRESSION_LEFT_ID', 1);
 	define('ZBX_EXPRESSION_SIMPLE_EXPRESSION_ID', 2);
