diff -Nur zabbix-1.4.7.org/frontends/php/include/classes/graph.inc.php zabbix-1.4.7/frontends/php/include/classes/graph.inc.php
--- zabbix-1.4.7.org/frontends/php/include/classes/graph.inc.php	2011-01-03 08:11:03.000000000 -0800
+++ zabbix-1.4.7/frontends/php/include/classes/graph.inc.php	2011-01-20 21:21:49.000000000 -0800
@@ -408,7 +408,7 @@
 				$fontnum = 4;
 			}
 			$x=$this->fullSizeX/2-ImageFontWidth($fontnum)*strlen($str)/2;
-			ImageString($this->im, $fontnum,$x,1, $str , $this->GetColor("Dark Red No Alpha"));
+			ImageTTFText($this->im, $fontnum+6, 0, $x+10, 13, $this->GetColor("Dark Red No Alpha"), "/usr/share/fonts/ja/TrueType/kochi-gothic-subst.ttf", $str);
 		}
 
 		function setHeader($header)
@@ -644,11 +644,12 @@
 				ImageFilledRectangle($this->im,$this->shiftXleft,$this->sizeY+$this->shiftY+62+12*$i,$this->shiftXleft+5,$this->sizeY+$this->shiftY+5+62+12*$i,$color);
 				ImageRectangle($this->im,$this->shiftXleft,$this->sizeY+$this->shiftY+62+12*$i,$this->shiftXleft+5,$this->sizeY+$this->shiftY+5+62+12*$i,$this->GetColor("Black No Alpha"));
 
-				ImageString($this->im, 2,
-					$this->shiftXleft+9,
-					$this->sizeY+$this->shiftY+(62-5)+12*$i,
-					$str,
-					$this->GetColor("Black No Alpha"));
+				ImageTTFText($this->im, 8, 0,
+ 					$this->shiftXleft+9,
+					$this->sizeY+$this->shiftY+(62-5)+12*$i+11,
+					$this->GetColor("Black No Alpha"),
+					"/usr/share/fonts/ja/TrueType/kochi-gothic-subst.ttf",
+					$str);
 			}
 
 			if($this->sizeY < 120) return;
diff -Nur zabbix-1.4.7.org/frontends/php/map.php zabbix-1.4.7/frontends/php/map.php
--- zabbix-1.4.7.org/frontends/php/map.php	2011-01-03 08:11:03.000000000 -0800
+++ zabbix-1.4.7/frontends/php/map.php	2011-01-20 21:25:04.000000000 -0800
@@ -103,7 +103,7 @@
 	else
 	{
 		$x=imagesx($im)/2-ImageFontWidth(4)*strlen($name)/2;
-		ImageString($im, 4,$x,1, $name , $darkred);
+		ImageTTFText($im, 10, 0, $x+10, 14, $darkred, "/usr/share/fonts/ja/TrueType/kochi-gothic-subst.ttf", $name);
 	}
 	unset($db_image);
 
@@ -251,7 +251,8 @@
 				$x_label-2, $y_label,
 				$x_label+ImageFontWidth(2)*strlen($label_line), $y_label+ImageFontHeight(2),
 				$white);
-			ImageString($im, 2, $x_label, $y_label, $label_line,$label_color);
+			ImageTTFText($im, 9, 0, $x_label+2, $y_label+10, $label_color,"/usr/share/fonts/ja/TrueType/kochi-gothic-subst.ttf", $label_line);
+
 		}
 
 		if($info_line!="")
@@ -260,7 +261,8 @@
 				$x_info-2, $y_info,
 				$x_info+ImageFontWidth(2)*strlen($info_line), $y_info+ImageFontHeight(2),
 				$white);
-			ImageString($im, 2, $x_info, $y_info, $info_line,$color);
+			ImageTTFText($im, 8, 0, $x_info+9, $y_info+10, $color, "/usr/share/fonts/ja/TrueType/kochi-gothic-subst.ttf", $info_line);
+
 		}
 	}
 
