diff -urN zabbix-1.8.17.orig/create/schema/ibm_db2.sql zabbix-1.8.17/create/schema/ibm_db2.sql
--- zabbix-1.8.17.orig/create/schema/ibm_db2.sql	2013-07-26 17:51:31.000000000 +0900
+++ zabbix-1.8.17/create/schema/ibm_db2.sql	2013-07-26 19:56:21.708999105 +0900
@@ -762,7 +762,7 @@
 	sendto		varchar(100)		DEFAULT ''	NOT NULL,
 	active		integer		DEFAULT '0'	NOT NULL,
 	severity		integer		DEFAULT '63'	NOT NULL,
-	period		varchar(100)		DEFAULT '1-7,00:00-23:59'	NOT NULL,
+	period		varchar(100)		DEFAULT '1-7,00:00-24:00'	NOT NULL,
 	PRIMARY KEY (mediaid)
 );
 CREATE INDEX media_1 on media (userid);
diff -urN zabbix-1.8.17.orig/create/schema/mysql.sql zabbix-1.8.17/create/schema/mysql.sql
--- zabbix-1.8.17.orig/create/schema/mysql.sql	2013-07-26 17:51:31.000000000 +0900
+++ zabbix-1.8.17/create/schema/mysql.sql	2013-07-26 19:56:44.565001313 +0900
@@ -762,7 +762,7 @@
 	sendto		varchar(100)		DEFAULT ''	NOT NULL,
 	active		integer		DEFAULT '0'	NOT NULL,
 	severity		integer		DEFAULT '63'	NOT NULL,
-	period		varchar(100)		DEFAULT '1-7,00:00-23:59'	NOT NULL,
+	period		varchar(100)		DEFAULT '1-7,00:00-24:00'	NOT NULL,
 	PRIMARY KEY (mediaid)
 ) ENGINE=InnoDB;
 CREATE INDEX media_1 on media (userid);
diff -urN zabbix-1.8.17.orig/create/schema/oracle.sql zabbix-1.8.17/create/schema/oracle.sql
--- zabbix-1.8.17.orig/create/schema/oracle.sql	2013-07-26 17:51:31.000000000 +0900
+++ zabbix-1.8.17/create/schema/oracle.sql	2013-07-26 19:57:27.981001616 +0900
@@ -1096,7 +1096,7 @@
 	sendto		nvarchar2(100)		DEFAULT '',
 	active		number(10)		DEFAULT '0'	NOT NULL,
 	severity		number(10)		DEFAULT '63'	NOT NULL,
-	period		nvarchar2(100)		DEFAULT '1-7,00:00-23:59',
+	period		nvarchar2(100)		DEFAULT '1-7,00:00-24:00',
 	PRIMARY KEY (mediaid)
 )
 /
diff -urN zabbix-1.8.17.orig/create/schema/postgresql.sql zabbix-1.8.17/create/schema/postgresql.sql
--- zabbix-1.8.17.orig/create/schema/postgresql.sql	2013-07-26 17:51:31.000000000 +0900
+++ zabbix-1.8.17/create/schema/postgresql.sql	2013-07-26 19:57:45.564997600 +0900
@@ -762,7 +762,7 @@
 	sendto		varchar(100)		DEFAULT ''	NOT NULL,
 	active		integer		DEFAULT '0'	NOT NULL,
 	severity		integer		DEFAULT '63'	NOT NULL,
-	period		varchar(100)		DEFAULT '1-7,00:00-23:59'	NOT NULL,
+	period		varchar(100)		DEFAULT '1-7,00:00-24:00'	NOT NULL,
 	PRIMARY KEY (mediaid)
 ) with OIDS;
 CREATE INDEX media_1 on media (userid);
diff -urN zabbix-1.8.17.orig/create/schema/sqlite.sql zabbix-1.8.17/create/schema/sqlite.sql
--- zabbix-1.8.17.orig/create/schema/sqlite.sql	2013-07-26 17:51:31.000000000 +0900
+++ zabbix-1.8.17/create/schema/sqlite.sql	2013-07-26 19:58:02.021002506 +0900
@@ -758,7 +758,7 @@
 	sendto		varchar(100)		DEFAULT ''	NOT NULL,
 	active		integer		DEFAULT '0'	NOT NULL,
 	severity		integer		DEFAULT '63'	NOT NULL,
-	period		varchar(100)		DEFAULT '1-7,00:00-23:59'	NOT NULL,
+	period		varchar(100)		DEFAULT '1-7,00:00-24:00'	NOT NULL,
 	PRIMARY KEY (mediaid)
 );
 CREATE INDEX media_1 on media (userid);
diff -urN zabbix-1.8.17.orig/frontends/php/actionconf.php zabbix-1.8.17/frontends/php/actionconf.php
--- zabbix-1.8.17.orig/frontends/php/actionconf.php	2013-07-26 17:50:53.000000000 +0900
+++ zabbix-1.8.17/frontends/php/actionconf.php	2013-07-26 19:59:05.452998542 +0900
@@ -688,7 +688,7 @@
 					$rowCondition[] = $cmbCondVal;
 					break;
 				case CONDITION_TYPE_TIME_PERIOD:
-					$rowCondition[] = new CTextBox('new_condition[value]', "1-7,00:00-23:59", 40);
+					$rowCondition[] = new CTextBox('new_condition[value]', "1-7,00:00-24:00", 40);
 					break;
 				case CONDITION_TYPE_TRIGGER_SEVERITY:
 					$cmbCondVal = new CComboBox('new_condition[value]');
diff -urN zabbix-1.8.17.orig/frontends/php/include/forms.inc.php zabbix-1.8.17/frontends/php/include/forms.inc.php
--- zabbix-1.8.17.orig/frontends/php/include/forms.inc.php	2013-07-26 17:50:52.000000000 +0900
+++ zabbix-1.8.17/frontends/php/include/forms.inc.php	2013-07-26 20:00:15.406995519 +0900
@@ -1987,7 +1987,7 @@
 						S_DELAY, SPACE,
 						new CNumericBox('new_delay_flex[delay]','50',5),
 						S_PERIOD, SPACE,
-						new CTextBox('new_delay_flex[period]','1-7,00:00-23:59',27), BR(),
+						new CTextBox('new_delay_flex[period]','1-7,00:00-24:00',27), BR(),
 						new CButton('add_delay_flex',S_ADD)
 					),'form_row_r')), 'new');
 		$row->setAttribute('id', 'row_new_delay_flex');
@@ -2316,7 +2316,7 @@
 										S_DELAY, SPACE,
 										new CNumericBox("new_delay_flex[delay]","50",5),
 										S_PERIOD, SPACE,
-										new CTextBox("new_delay_flex[period]","1-7,00:00-23:59",27), BR(),
+										new CTextBox("new_delay_flex[period]","1-7,00:00-24:00",27), BR(),
 										new CButton("add_delay_flex",S_ADD)
 									));
 		$new_delay_flex_el->setAttribute('id', 'new_delay_flex_el');
diff -urN zabbix-1.8.17.orig/frontends/php/include/schema.inc.php zabbix-1.8.17/frontends/php/include/schema.inc.php
--- zabbix-1.8.17.orig/frontends/php/include/schema.inc.php	2013-07-26 17:50:52.000000000 +0900
+++ zabbix-1.8.17/frontends/php/include/schema.inc.php	2013-07-26 20:01:28.734001072 +0900
@@ -3460,7 +3460,7 @@
 				'null' => false,
 				'type' => DB::FIELD_TYPE_CHAR,
 				'length' => 100,
-				'default' => '1-7,00:00-23:59',
+				'default' => '1-7,00:00-24:00',
 			),
 		),
 	),
diff -urN zabbix-1.8.17.orig/frontends/php/popup_media.php zabbix-1.8.17/frontends/php/popup_media.php
--- zabbix-1.8.17.orig/frontends/php/popup_media.php	2013-07-26 17:50:53.000000000 +0900
+++ zabbix-1.8.17/frontends/php/popup_media.php	2013-07-26 20:02:08.149998620 +0900
@@ -100,7 +100,7 @@
 	$sendto		=  get_request('sendto','');
 	$mediatypeid	= get_request('mediatypeid',0);
 	$active		= get_request('active',0);
-	$period		= get_request('period','1-7,00:00-23:59');
+	$period		= get_request('period','1-7,00:00-24:00');
 
 
 	$frmMedia = new CFormTable(S_NEW_MEDIA);
diff -urN zabbix-1.8.17.orig/src/libs/zbxdbhigh/dbschema.c zabbix-1.8.17/src/libs/zbxdbhigh/dbschema.c
--- zabbix-1.8.17.orig/src/libs/zbxdbhigh/dbschema.c	2013-07-26 17:51:31.000000000 +0900
+++ zabbix-1.8.17/src/libs/zbxdbhigh/dbschema.c	2013-07-26 20:03:15.277998662 +0900
@@ -1939,7 +1939,7 @@
  sendto varchar(100) DEFAULT '' NOT NULL,\
  active integer DEFAULT '0' NOT NULL,\
  severity integer DEFAULT '63' NOT NULL,\
- period varchar(100) DEFAULT '1-7,00:00-23:59' NOT NULL,\
+ period varchar(100) DEFAULT '1-7,00:00-24:00' NOT NULL,\
  PRIMARY KEY (mediaid)\
 );\
 CREATE INDEX media_1 on media (userid);\
diff -urN zabbix-1.8.17.orig/upgrades/dbpatches/1.8/oracle/patch.sql zabbix-1.8.17/upgrades/dbpatches/1.8/oracle/patch.sql
--- zabbix-1.8.17.orig/upgrades/dbpatches/1.8/oracle/patch.sql	2013-07-26 17:51:34.000000000 +0900
+++ zabbix-1.8.17/upgrades/dbpatches/1.8/oracle/patch.sql	2013-07-26 20:03:57.673998586 +0900
@@ -585,7 +585,7 @@
 alter table mappings modify newvalue                nvarchar2(64)           DEFAULT '';
 
 alter table media modify sendto          nvarchar2(100)          DEFAULT '';
-alter table media modify period          nvarchar2(100)          DEFAULT '1-7,00:00-23:59';
+alter table media modify period          nvarchar2(100)          DEFAULT '1-7,00:00-24:00';
 
 alter table media_type modify description             nvarchar2(100)          DEFAULT '';
 alter table media_type modify smtp_server             nvarchar2(255)          DEFAULT '';
